### PR TITLE
bpo-37621: Don't emit NEWLINE tokens on blank line continuations

### PR DESCRIPTION
**Pull request by miedzinski**
https://bugs.python.org/issue37621

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Our records indicate we have not received your CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Review by asottile — COMMENTED** [review]
feels correct to me -- mind adding some tests for this?

**Review by asottile — APPROVED** [review]
is there any measurable side-effects on the C tokenizer? not sure -- tests look good though

**Comment by miedzinski** [comment]
Yes, just found one. Currently (pre-PR)

```python
def f():
    \
  x = 1
  return x
```

throws an IndentationError (unmatched dedent) on line 4, because line 3 has indentation level from line 2. IMO indent from line 2 should be ignored, because it's empty.

File: Lib/test/test_tokenize.py [hunk 1]
```diff
@@ -964,6 +964,22 @@ def test_linecont(self):
...
+  return x""", """\
+    NAME       'def'         (1, 0) (1, 3)
+    NAME       'f'           (1, 4) (1, 5)
+    OP         '('           (1, 5) (1, 6)
+    NAME       'x'           (1, 6) (1, 7)
+    OP         ')'           (1, 7) (1, 8)
+    OP         ':'           (1, 8) (1, 9)
+    NEWLINE    '\\n'          (1, 9) (1, 10)
+    NL         '\\n'          (2, 5) (2, 6)
+    INDENT     '  '          (3, 0) (3, 2)
```
Comment (miedzinski) [inline Lib/test/test_tokenize.py:978]: Before PR the indent would be 4 spaces, not 2.

**Comment by miedzinski** [comment]
Now this code is accepted, because we calculate indentation _after_ line continuation and then push it to the stack. C and Python tokenizers should behave the same.

**Comment by lysnikolaou** [comment]
This is no longer relevant after the changes to the tokenizer in 3.12.

**Comment by asottile** [comment]
@lysnikolaou this change also was fixing the c tokenizer

**Comment by lysnikolaou** [comment]
The C tokenizer tokenizes the file attached to issue correctly as far as I can tell.

```python3
cpython on  main [$?] via C v15.0.0-clang via 🐍 pyenv 3.11.3 
❯ cat linecont.py
\

\

\


cpython on  main [$?] via C v15.0.0-clang via 🐍 pyenv 3.11.3 
❯ ./python.exe -m tokenize linecont.py
0,0-0,0:            ENCODING       'utf-8'        
2,0-2,1:            NL             '\n'           
4,0-4,1:            NL             '\n'           
6,0-6,1:            NL             '\n'           
7,0-7,0:            ENDMARKER      ''             
```

**Pull request closed without merging**